devtools/src/shared/rpc/custom-values.ts

```diff
@@ -74,14 +273,59 @@ function restoreClonedValue(value: NonSerializableValue) {
     try {
       return BigInt(value.value)
     } catch (err) {
-      console.warn(`[🍹]: Invalid bigint value: ${value.value}`, err)
-      return 0n
+      return new SerializationError(`Invalid bigint value: ${value.value}`, err)
     }
+  } else if (value.__type === 'regexp') {
+    try {
+      const regexpValue = value as NonSerializableValue_RegExp
+      const { source, flags } = regexpValue.value
```

Comment:
All these `as ...` cast need to be removed, they aren't needed now. Here is an example. Apply it to all cases.
```suggestion
      const { source, flags } = value.value
```